cassandra/cluster.py

```diff
@@ -1003,6 +1003,11 @@ def default_retry_policy(self, policy):
     load the configuration and certificates.
```

Comment:
I'm not sure if the second commit is related...

I'm not sure which implementation you are basing your implementation off, but in gocql it is possible to disable the _shard-aware port_, not _shard-awareness_ in its entirety - and I think this is what should be implemented here. The reason for that switch was that if the shard-aware port didn't work properly then the user might want to switch it off and continue using the regular port, __but still have shard-awareness and one connection per shard__.

Reply:
It's actually coming from a few real users, that the automatic shard awareness was causing them problems, they had strong machines (with lot of cpus/shards) and the amount of connections being opened was causing them some issue.

and there was no way to disable it. they switched to datastax driver cause of that.